### ISSUE TITLE
[Bug]: Latest Nightly Build Doesn't Disable SIP

### ISSUE THREAD

**Issue by ylluminarious** [dortania/OpenCore-Legacy-Patcher#513]
### Machine Model

MacPro5,1

### Application Version

Latest Build (nightly)

### Application Variant

GUI (Graphical User Interface)

### What versions of macOS are you seeing the problem on?

macOS 11, Big Sur

### Where does this issue happen?

Within macOS (After Logging in)

### What is the Isssue?

On the latest nightly build (`8f98cac`), I disabled SIP in the GUI settings as follows:

![gui_settings](https://user-images.githubusercontent.com/4912973/134848764-6e4e839c-e2d9-43bd-a12b-26e3924d479d.png)

However, I still get the following output from `csrutil status` (showing that it was not fully disabled):

```
System Integrity Protection status: unknown (Custom Configuration).

Configuration:
	Apple Internal: disabled
	Kext Signing: disabled
	Filesystem Protections: disabled
	Debugging Restrictions: enabled
	DTrace Restrictions: enabled
	NVRAM Protections: enabled
	BaseSystem Verification: enabled

This is an unsupported configuration, likely to break in the future and leave your machine in an unknown state.
```

This has also happened on another machine using the latest nightly build (`8f98cac`). Any ideas on how to fix this?

**Comment by derelvis** [comment]
Same for me in release 0.25.
Btw. your screenshot is misleading cause it is not showing it unchecked ;-)

**Comment by ylluminarious** [comment]
@derelvis I don't understand your remark about the screenshot - if it were unchecked, then SIP wouldn't be disabled at all. My goal is to disable SIP.

**Comment by derelvis** [comment]
@ylluminarious Sorry, you are right (in your case). For me it is the other way round: wanted to get SIP enabled and therfor unchecked the box. But it did not turn on. Maybe the checkbox does not work anymore at all?

**Comment by jgleigh** [comment]
There are issues with the checkboxes for SIP and SBM in the GUI. Use the TUI and it will work as expected until this gets fixed.

**Comment by ylluminarious** [comment]
@khronokernel I tried @jgleigh's suggestion above for using the TUI app instead of the GUI app, and I ran the latest bleeding edge again. I tried running `OCLP-CLI` from [these build artifacts,](https://github.com/dortania/OpenCore-Legacy-Patcher/runs/3724084559) but all that did was print a vague message about changing payload paths (I unfortunately didn't copy the output). Once I realized this was not the actual TUI app, but a CLI program instead, I then tried running `OpenCore-Patcher-TUI-Offline.app` from the same commit, and rebooted.

Unfortunately, when my computer booted up again, it appears that something wonky occurred, in that many of my applications have lost authentication-based info (had to re-add/sign-in to email accounts on email apps; reinstall Chrome extensions; lost other settings; iMessage was no longer working; among many other apps, etc.). 

Simply put, lots of things wanted me to sign in again, which was a strong indicator that my old hardware ID was changed somehow. I've seen this kind of behavior before when some hardware changes occur.

Do you know what could have caused this issue? @ylluminate suggested that I talk to you about this in order to try to figure out what's going on since perhaps a change in OCLP might be the root of the problem, since I haven't changed anything else that could have caused this. Any help (especially a way to reverse this if possible) would be much appreciated 😄

**Comment by jgleigh** [comment]
There's a few different things going on here.

The GUI doesn't correctly look at the SIP and SBM checkboxes. It always uses the settings that are pre-determined and doesn't allow them to be overridden.

The TUI does allow you enable SIP and SBM, but a recent change in the nightly builds doesn't fully disable SIP when it's set to be disabled. It partially disables it which is what you saw.

There was also a recent change (reversion actually) for MacPros to go back to the MacPro7,1 SMBIOS setting instead of the iMacPro1,1 setting. This is probably what caused the need to reset your settings. I saw something similar and had to relog into iCloud, iMessage, etc.

**Comment by khronokernel** [comment]
Duplicate: https://github.com/dortania/OpenCore-Legacy-Patcher/issues/452

**Comment by ylluminarious** [comment]
@jgleigh Thanks, that makes a lot of sense and clarifies some of the confusion I'm seeing here. Do you think the SMBIOS setting will again be changed at some future point? Now that I'm signed back into everything, I don't necessarily want to have to deal with yet another hardware reset... unless it would reset my hardware ID back to what it previously was.

**Comment by ylluminate** [comment]
The MacPro7,1 to iMacPro1,1 move made a lot of sense @khronokernel, what's the new rationale of going back and/or can we control this vs having it defaulted this way?

**Comment by khronokernel** [comment]
> what's the new rationale of going back

GPUs broke

> can we control this vs having it defaulted this way?

Patcher Settings

**Comment by ylluminate** [comment]
Thanks - I know we went through a lot with the VEGA64 situation and I suppose there were a lot more issues then going on as well. Sorry to hear that - I'm sure it's been a headache.
